packages/o/objfw/xmake.lua

```diff
@@ -12,13 +12,94 @@ package("objfw")
+    add_configs("codepage-858", { description = "Enable codepage 858 support.", default = true, type = "boolean" })
+    add_configs("iso-8859-2", { description = "Enable ISO-8859-2 support.", default = true, type = "boolean" })
+    add_configs("iso-8859-3", { description = "Enable ISO-8859-3 support.", default = true, type = "boolean" })
+    add_configs("iso-8859-15", { description = "Enable ISO-8859-15 support.", default = true, type = "boolean" })
+    add_configs("koi8-r", { description = "Enable KOI8-R support.", default = true, type = "boolean" })
+    add_configs("koi8-u", { description = "Enable KOI8-U support.", default = true, type = "boolean" })
+    add_configs("mac-roman", { description = "Enable Mac Roman encoding support.", default = true, type = "boolean" })
+    add_configs("windows-1251", { description = "Enable windows 1251 support.", default = true, type = "boolean" })
+    add_configs("windows-1252", { description = "Enable windows 1252 support.", default = true, type = "boolean" })
+
+    add_configs("threads", { description = "Enable threads.", default = true, type = "boolean" })
+    add_configs("compiler-tls", { description = "Enable compiler thread local storage (TLS).", default = true, type = "boolean" })
+    add_configs("files", { description = "Enable files.", default = true, type = "boolean" })
+    add_configs("sockets", { description = "Enable sockets.", default = true, type = "boolean" })
+
+    add_configs("arc", { description = "Enable Automatic Reference Counting (ARC) support.", default = true, type = "boolean" })
+
+    on_install("linux", "macosx", "cygwin", "mingw", function (package)
+        local configs = {}
+
+        local tls = package:config("tls")
+        tls = tls == true and "yes" or tls == false and "no" or tls
+        table.insert(configs, "--enable-tls=" .. tls)
+
+        local function config(cfg)
+            if package:config(cfg) then
+                table.insert(configs, "--enable-" .. cfg)
+            else
+                table.insert(configs, "--disable-" .. cfg)
+            end
+        end
+
         table.insert(configs, "--enable-shared=" .. (package:config("shared") and "yes" or "no"))
-        if package:debug() then
+        table.insert(configs, "--enable-static=" .. (package:config("shared") and "no" or "yes"))
+
+        config("threads")
+        config("compiler-tls")
+        config("files")
+        config("sockets")
+        config("runtime")
+        config("seluid24")
+        config("unicode-tables")
+        config("rpath")
+
+        config("codepage-437")
+        config("codepage-858")
+        config("codepage-850")
+        config("iso-8859-3")
+        config("iso-8859-2")
+        config("koi8-r")
+        config("koi8-u")
+        config("iso-8859-15")
+        config("mac-roman")
+        config("windows-1251")
+        config("windows-1252")
```

Comment:
you can for-each package:configs()

see

https://github.com/xmake-io/xmake-repo/blob/2f0b2cb3c771dd81121d05bc8df68516c749ff87/packages/f/ffmpeg/xmake.lua#L122